### PR TITLE
Setup project to accomodate client/server apps

### PR DESCRIPTION
**Pull request by dayllanmaza**
- Re-stucture repo into two folders (client/server). 
- Move js app into client and setup lighttpd to server static html and the
future to be api. 
- Modify docker-compose to provide needed containers
for the api development

Bug: T185464, T185461

### CODE REVIEW

**Review by MusikAnimal — APPROVED** [review]
Doesn't look like there's much for code changes here, rather just moving things around. I did get it working locally, runs great! The hello API is also waving back, so looks good to me.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
   build:
     image: docker-registry.tools.wmflabs.org/toollabs-nodejs-web
-    working_dir: /code
+    working_dir: /code/client/
     command:
       - ./bin/watch
     volumes:
-      - ./:/code:cached
+      - ./:/code
+  composer:
```
Comment (davidbarratt) [inline docker-compose.yml:19]: Is composer on toolforge? or is it in the `docker-registry.tools.wmflabs.org/toollabs-php-web` image?

File: etc/lighttpd/lighttpd.conf [hunk 2]
```diff
@@ -3,18 +3,18 @@ server.modules = (
 	"mod_alias",
 	"mod_compress",
  	"mod_redirect",
+ 	"mod_fastcgi",
 	"mod_rewrite",
 )
 
-server.document-root        = "/var/www/html"
+server.document-root        = "/var/www/client/html"
```
Comment (davidbarratt) [inline etc/lighttpd/lighttpd.conf:10]: why not have a single `html` root?

Comment (davidbarratt) [reply etc/lighttpd/lighttpd.conf:10]: in other words, you'd have `./html` which would contain both `index.html` as well as `api/index.php`? If that's a bad idea then we shouldn't do it, but it seems confusing that there are two webroots.

File: docker-compose.yml (hunk 1 above)
Comment (dayllanmaza) [reply docker-compose.yml:19]: now that you mentioned it, toollabs-php-web image has composer. We could move this to be a command on web instead. Although, I kinda like it separate

File: etc/lighttpd/lighttpd.conf (hunk 2 above)
Comment (dayllanmaza) [reply etc/lighttpd/lighttpd.conf:10]: Initially I went for that but `build` re-creates the `html` folder and `web` was mounting it before `build` was able to do its work. So after `build` was done, whatever I mounted on `html` was removed. An alternative is to wait for all of it to be done and symlink the folders into the document root. It doens't bother me this way and it works just fine but if you wanna change it I don't mind it at all

**Pull request merged**